src/platform/site-wide/user-nav/containers/Main.jsx

```diff
@@ -70,8 +73,16 @@ export class Main extends React.Component {
   }
 
   checkLoggedInStatus = () => {
+    ssoKeepAliveSession();
+    const canCallSSO = this.props.useSSOe && !environment.isLocalhost();
```

Comment:
given that there's not a quick way to verify auto-login/logout locally without modifying code, disabling it for localhost by default so as to not cause errors for sessions

Reply:
I was grappling with this and got stuck, but I'm open to any ideas. The problem is two-fold:

1. Since we don't own that endpoint, we get CORS issues from localhost. I was able to work around that with the mock api call.
2. However, that is just always defaulting to returning a response that indicates an active session. 

The problem with the mock call with respect to auto login/logout is it can't accurately simulate the logic here. I.e., any time someone hasn't logged in locally, `ssoKeepAliveSession()` returns an "active" session, and this would attempt to redirect them to log in. Conversely, if you were to flip it to false and the user _was_ logged in, it would log them out. Which felt overly cumbersome to most developers for testing something locally that they probably would rather not worry about.

The only idea that I toyed around with without committing to was some sort of popup devtools for local development, where you could toggle whether or not the keep-alive call returns an active SSO session. I nixed it cuz if a developer didn't know it was changeable, this logic would be annoying. Thinking about it now, you could add an additional flag to toggle whether or not to have the auto login/logout enabled and have that be part of the `canCallSSO` logic, something like

```js
const can CallSOO = this.props.useSSOe && (environment.isLocalhost() && checkAutoSSO)
```

does that seem worthwhile down the road? Idk if there's other things that could be useful to include there for local development outside of this single feature, but the general inspiration for it came from this [blog post](https://kentcdodds.com/blog/make-your-own-dev-tools)

Reply:
Custom dev tools is an interesting idea. Might be something the platform can think about supporting in a more standard way. But for now, I think defaulting to turning it off for local dev is the safe call so no one will be caught off guard by it.

Reply:
Yeah, you could also check a value in localStorage to serve as a flag during local dev